lib/cli/src/commands/publish.rs

```diff
@@ -0,0 +1,647 @@
+                key_type_identifier: row.get(4)?,
+                public_key_id: row.get(5)?,
+            })
+        })?
+        .next();
+
+    if let Some(res) = result {
+        Ok(res?)
+    } else {
+        Err(anyhow!("No active key found"))
+    }
+}
+
+mod interfaces {
+
+    use rusqlite::{params, Connection, TransactionBehavior};
+
+    pub const WASM_INTERFACE_EXISTENCE_CHECK: &str =
+        include_str!("./sql/wasm_interface_existence_check.sql");
+    pub const INSERT_WASM_INTERFACE: &str = include_str!("./sql/insert_interface.sql");
+    pub const GET_WASM_INTERFACE: &str = include_str!("./sql/get_interface.sql");
+
+    pub fn interface_exists(
+        conn: &mut Connection,
+        interface_name: &str,
+        version: &str,
+    ) -> anyhow::Result<bool> {
+        let mut stmt = conn.prepare(WASM_INTERFACE_EXISTENCE_CHECK)?;
+        Ok(stmt.exists(params![interface_name, version])?)
+    }
+
+    pub fn load_interface_from_db(
+        conn: &mut Connection,
+        interface_name: &str,
+        version: &str,
+    ) -> anyhow::Result<wasmer_wasm_interface::Interface> {
+        let mut stmt = conn.prepare(GET_WASM_INTERFACE)?;
+        let interface_string: String =
+            stmt.query_row(params![interface_name, version], |row| row.get(0))?;
+
+        wasmer_wasm_interface::parser::parse_interface(&interface_string).map_err(|e| {
+            anyhow!(
+                "Failed to parse interface {} version {} in database: {}",
+                interface_name,
+                version,
+                e
+            )
+        })
+    }
+
+    pub fn import_interface(
+        conn: &mut Connection,
+        interface_name: &str,
+        version: &str,
+        content: &str,
+    ) -> anyhow::Result<()> {
+        // fail if we already have this interface
+        {
+            let mut key_check = conn.prepare(WASM_INTERFACE_EXISTENCE_CHECK)?;
+            let result = key_check.exists(params![interface_name, version])?;
+
+            if result {
+                return Err(anyhow!(
+                    "Interface {}, version {} already exists",
+                    interface_name,
+                    version
+                ));
+            }
+        }
+
+        let tx = conn.transaction_with_behavior(TransactionBehavior::Immediate)?;
+        let time_string = get_current_time_in_format().expect("Could not get current time");
+
+        log::debug!("Adding interface {:?} {:?}", interface_name, version);
+        tx.execute(
+            INSERT_WASM_INTERFACE,
+            params![interface_name, version, time_string, content],
+        )?;
+
+        tx.commit()?;
+        Ok(())
+    }
+
+    /// Gets the current time in our standard format
+    pub fn get_current_time_in_format() -> Option<String> {
+        use time::format_description::well_known::Rfc3339;
+        let cur_time = time::OffsetDateTime::now_utc();
+        cur_time.format(&Rfc3339).ok()
+    }
+}
+
+mod validate {
```

Comment:
I would like to keep it here, so that we have all of the publishing logic in one file (init.rs, login.rs, publish.rs). But it should be in a separate module so it doesn't interfere with anything else.